ffi/diplomat/src/displaynames.rs

```diff
@@ -0,0 +1,141 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::errors::ffi::ICU4XError;
+    use crate::locale::ffi::ICU4XLocale;
+    use crate::provider::ffi::ICU4XDataProvider;
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatWriteable;
+    #[allow(unused_imports)] // feature-specific
+    use icu_displaynames::options::{DisplayNamesOptions, Fallback, LanguageDisplay};
+    use icu_displaynames::{LanguageDisplayNames, RegionDisplayNames};
+    use writeable::Writeable;
+
+    //  FFI version of `LanguageDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::LanguageDisplayNames, Struct)]
+    pub struct ICU4XLanguageDisplayNames(pub LanguageDisplayNames);
+
+    //  FFI version of `RegionDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::RegionDisplayNames, Struct)]
+    pub struct ICU4XRegionDisplayNames(pub RegionDisplayNames);
+
+    //  FFI version of `DisplayNamesOptions`.
+    #[diplomat::rust_link(icu::displaynames::options::DisplayNamesOptions, Struct)]
+    pub struct ICU4XDisplayNamesOptions {
+        /// The optional formatting style to use for display name.
+        pub style: ICU4XDisplayNamesStyle,
+        /// The fallback return when the system does not have the
+        /// requested display name, defaults to "code".
+        pub fallback: ICU4XDisplayNamesFallback,
+        /// The language display kind, defaults to "dialect".
+        pub language_display: ICU4XLanguageDisplay,
+    }
+
+    // FFI version of `Style` option for the display names.
+    #[diplomat::rust_link(icu::displaynames::options::Style, Enum)]
+    pub enum ICU4XDisplayNamesStyle {
+        Auto,
+        Narrow,
+        Short,
+        Long,
+        Menu,
+    }
+
+    // FFI version of `Fallback` option for the display names.
+    #[diplomat::rust_link(icu::displaynames::options::Fallback, Enum)]
+    #[diplomat::enum_convert(Fallback, needs_wildcard)]
+    pub enum ICU4XDisplayNamesFallback {
+        Code,
+        None,
+    }
+
+    // FFI version of `LanguageDisplay`.
+    #[diplomat::rust_link(icu::displaynames::options::LanguageDisplay, Enum)]
+    #[diplomat::enum_convert(LanguageDisplay, needs_wildcard)]
+    pub enum ICU4XLanguageDisplay {
```

Comment:
Thought: I wonder if this enum is sufficiently namespaced. It's probably fine.

Reply:
I was also wondering that and came to the same conclusion

Reply:
Do you mean the enum name should be starting with ICU4XDisplayNames?